jujugui/static/gui/src/app/components/post-deployment/post-deployment.js

```diff
@@ -106,6 +121,84 @@ class PostDeployment extends React.Component {
     }
   }
 
+  /**
+    Callback for when the post-deployment script has been fetched.
+
+    @param {String} error Error from the API.
+    @param {String} postDeploymentScript The content of the script.
+  */
+  _getPostDeploymentScriptCallback(error, postDeploymentScript) {
+    if (error) {
+      console.error(error);
+    }
+    this.setState({
+      postDeploymentScript: postDeploymentScript
+    });
+  }
+
+  /**
+    Render a button for executing the post-deployment script if one exists.
+
+    @return {Object} The button if required, otherwise nothing.
+  */
+  _renderPostDeploymentScriptButton() {
+    const showPostDeploymentScript = this.props.showPostDeploymentScript;
+    if (showPostDeploymentScript && this.state.postDeploymentScript) {
+      return (<div>
+        <GenericButton
+          action={this._executePostDeploymentScript.bind(this)}>
+            Execute post-deployment script
+        </GenericButton>
+      </div>);
+    }
+  }
+
+  /**
+    Execute the post-deployment script by opening the terminal with the script
+    as a payload.
+  */
+  _executePostDeploymentScript() {
+    const scriptLines = this.state.postDeploymentScript.split('\n');
+    this.props.changeState({terminal: scriptLines});
+  }
+
+  /**
+    Callback for when the post-deployment script has been fetched.
+
+    @param {String} error Error from the API.
+    @param {String} postDeploymentScript The content of the script.
+  */
+  _getPostDeploymentScriptCallback(error, postDeploymentScript) {
+    if (error) {
+      console.error(error);
+    }
+    this.setState({
+      postDeploymentScript: postDeploymentScript
+    });
+  }
```

Comment:
Looks like you have this twice? Also at https://github.com/juju/juju-gui/pull/3761/files#diff-a447828941fdc011bc9d5215a53ab5b2R130